src/redis-cli.c

```diff
@@ -2065,6 +2067,7 @@ static void repl(void) {
 
     cliRefreshPrompt();
     while((line = linenoise(context ? config.prompt : "not connected> ")) != NULL) {
+        cliRefreshPrompt();
```

Comment:
it seems a bit excessive to call this on every loop.
maybe this should be done in `cliConnect`?
or alternatively, add a global that says the prompt needs a refresh and make this line conditional (in which case we can probably remove the call to refresh that's done right before this loop).

Reply:
thanks @oranagra , just updated the code. think it should be proper to update the promit right after we reset the dbnum, since after reconnect it will show this properly. How do you think?